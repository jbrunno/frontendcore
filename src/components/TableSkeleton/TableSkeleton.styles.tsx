import { Table } from '@mui/material';
import { styled } from '@mui/material/styles';

export const TableStyled = styled(Table)`
  margin-left: none;
  grid-column: span 13;
`;
