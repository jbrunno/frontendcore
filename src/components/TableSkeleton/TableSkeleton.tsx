import React from 'react';
import Skeleton from '../Skeleton/Skeleton';
import TableBody from '../TableBody/TableBody';
import TableCell from '../TableCell/TableCell';
import TableHead from '../TableHead/TableHead';
import TableRow from '../TableRow/TableRow';
import { TableStyled } from './TableSkeleton.styles';

type TableSkeletonProps = {
  rows: number;
};

export function TableSkeleton({ rows }: TableSkeletonProps) {
  return (
    <TableStyled>
      <TableHead>
        <TableRow>
          <TableCell>
            <Skeleton variant="rectangular" height={20} animation="wave" />
          </TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {Array.from({ length: rows }).map((_, index) => {
          const key = index;
          return (
            <TableRow key={key}>
              <TableCell>
                <Skeleton variant="text" />
              </TableCell>
            </TableRow>
          );
        })}
      </TableBody>
    </TableStyled>
  );
}

export default TableSkeleton;
