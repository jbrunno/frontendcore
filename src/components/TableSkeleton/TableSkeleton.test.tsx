import React from 'react';
import { render } from '@test-utils/component-testing';
import { TableSkeleton } from './TableSkeleton';

test('renders the TableSkeleton', () => {
  const { container } = render(<TableSkeleton rows={3} />);
  expect(container).toMatchSnapshot();
});
