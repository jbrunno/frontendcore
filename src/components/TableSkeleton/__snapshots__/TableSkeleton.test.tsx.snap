// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TableSkeleton 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: translateX(-100%);
    -moz-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  50% {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }

  100% {
    -webkit-transform: translateX(100%);
    -moz-transform: translateX(100%);
    -ms-transform: translateX(100%);
    transform: translateX(100%);
  }
}

@keyframes animation-1 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-1 {
  display: table;
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
  margin-left: none;
  grid-column: span 13;
}

.emotion-1 caption {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.125rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  padding: 1rem;
  color: #3B3B4199;
  text-align: left;
  caption-side: bottom;
}

.emotion-2 {
  display: table-header-group;
}

.emotion-3 {
  color: inherit;
  display: table-row;
  vertical-align: middle;
  outline: 0;
}

.emotion-3.MuiTableRow-hover:hover {
  background-color: #0000000A;
}

.emotion-3.Mui-selected {
  background-color: rgba(74, 49, 255, 0.08);
}

.emotion-3.Mui-selected:hover {
  background-color: rgba(74, 49, 255, 0.12);
}

.emotion-4 {
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.5rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  color: #3B3B41DE;
}

.emotion-5 {
  display: block;
  background-color: rgba(59, 59, 65, 0.11);
  height: 1.2em;
  position: relative;
  overflow: hidden;
  -webkit-mask-image: -webkit-radial-gradient(white, black);
}

.emotion-5::after {
  -webkit-animation: animation-0 1.6s linear 0.5s infinite;
  animation: animation-0 1.6s linear 0.5s infinite;
  background: linear-gradient(
            90deg,
            transparent,
            #0000000A,
            transparent
          );
  content: '';
  position: absolute;
  -webkit-transform: translateX(-100%);
  -moz-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
}

.emotion-6 {
  display: table-row-group;
}

.emotion-8 {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.125rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  color: #3B3B41DE;
}

.emotion-9 {
  display: block;
  background-color: rgba(59, 59, 65, 0.11);
  height: auto;
  margin-top: 0;
  margin-bottom: 0;
  transform-origin: 0 55%;
  -webkit-transform: scale(1, 0.60);
  -moz-transform: scale(1, 0.60);
  -ms-transform: scale(1, 0.60);
  transform: scale(1, 0.60);
  border-radius: 4px/6.7px;
  -webkit-animation: animation-1 1.5s ease-in-out 0.5s infinite;
  animation: animation-1 1.5s ease-in-out 0.5s infinite;
}

.emotion-9:empty:before {
  content: "\\00a0";
}

<div>
  <table
    class="MuiTable-root emotion-0 emotion-1"
  >
    <thead
      class="MuiTableHead-root emotion-2"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head emotion-3"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium emotion-4"
          scope="col"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave emotion-5"
            style="height: 20px;"
          />
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root emotion-6"
    >
      <tr
        class="MuiTableRow-root emotion-3"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-8"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-9"
          />
        </td>
      </tr>
      <tr
        class="MuiTableRow-root emotion-3"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-8"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-9"
          />
        </td>
      </tr>
      <tr
        class="MuiTableRow-root emotion-3"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-8"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse emotion-9"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
